src/openloco/windows/construction/StationTab.cpp

```diff
@@ -0,0 +1,438 @@
+        int16_t yPos = self->widgets[widx::image].top + self->y;
+
+        if (_byte_1136063 & (1 << 7))
+        {
+            auto airportObj = objectmgr::get<airport_object>(_lastSelectedStationType);
+
+            auto imageId = gfx::recolour(airportObj->var_08, companyColour);
+
+            gfx::draw_image(dpi, xPos, yPos, imageId);
+        }
+        else if (_byte_1136063 & (1 << 6))
+        {
+            auto dockObj = objectmgr::get<dock_object>(_lastSelectedStationType);
+
+            auto imageId = gfx::recolour(dockObj->var_08, companyColour);
+
+            gfx::draw_image(dpi, xPos, yPos, imageId);
+        }
+        else if (_trackType & (1 << 7))
+        {
+            auto roadStationObj = objectmgr::get<road_station_object>(_lastSelectedStationType);
+
+            auto imageId = gfx::recolour(roadStationObj->var_0C, companyColour);
+
+            gfx::draw_image(dpi, xPos, yPos, imageId);
+
+            auto colour = _byte_5045FA[companyColour];
+
+            if (!(roadStationObj->flags & road_station_flags::recolourable))
+            {
+                colour = 46;
+            }
+
+            imageId = gfx::recolour(imageId, colour) + 1;
+
+            gfx::draw_image(dpi, xPos, yPos, imageId);
+        }
+        else
+        {
+            auto trainStationObj = objectmgr::get<train_station_object>(_lastSelectedStationType);
+
+            auto imageId = gfx::recolour(trainStationObj->var_0E, companyColour);
+
+            gfx::draw_image(dpi, xPos, yPos, imageId);
+
+            auto colour = _byte_5045FA[companyColour];
+
+            if (!(trainStationObj->flags & train_station_flags::recolourable))
+            {
+                colour = 46;
+            }
+
+            imageId = gfx::recolour(imageId, colour) + 1;
+
+            gfx::draw_image(dpi, xPos, yPos, imageId);
+        }
+
+        if (_stationCost != 0x80000000 && _stationCost != 0)
+        {
+            xPos = self->x + 69;
+            yPos = self->widgets[widx::image].bottom + self->y + 4;
+
+            auto args = FormatArguments();
+            args.push<uint32_t>(_stationCost);
+
+            gfx::draw_string_centred(*dpi, xPos, yPos, colour::black, string_ids::build_cost, &args);
+        }
+
+        xPos = self->x + 3;
+        yPos = self->widgets[widx::image].bottom + self->y + 16;
+        auto width = self->width - 4;
+        gfx::draw_rect_inset(dpi, xPos, yPos, width, 1, self->colours[1], (1 << 5));
+
+        if (!(_byte_522096 & (1 << 3)))
+            return;
+
+        auto args = FormatArguments();
+
+        if (_dword_1135F70 == 0xFFFFFFFF)
+        {
+            args.push(string_ids::new_station);
+        }
+        else
+        {
+            auto station = stationmgr::get(_dword_1135F70);
+            args.push(station->name);
+            args.push(station->town);
+        }
+
+        stringmgr::format_string(&_stringFormatBuffer[0], string_ids::new_station_buffer, &args);
```

Comment:
This is from within `draw_string_centered_clipped` any reason you have broeken it out of the function? That function is used in a number of places we should keep the function call unless there is a reason to not to.

Reply:
No reason, I just saw `sub_494C36` and forgot to check if it has been implemented. I guess I may as well include this now, shall I raise a separate PR?